### PR TITLE
Make default and object_hook default params consistent with msgpack

### PR DESCRIPTION
**Pull request by tiagocoutinho**
After using *msgpack_numpy* 0.4.2 to patch *msgpack*, when I pass my custom `default/object_hook` to `msgpack.pack(b)/msgpack.unpack(b)` I loose the ability to pack/unpack numpy.

Example:
```python
import pickle
import numpy
import msgpack
import msgpack_numpy
msgpack_numpy.patch()

def encode_pickle(obj):
    return dict(__pickle__=1, as_str=pickle.dumps(obj))

data = numpy.arange(10)

# this works, obviously:
msgpack.packb(data)

# this raises TypeError: can't serialize array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
msgpack.packb(data, default=encode_pickle)
```

This PR proposes to fix that.

### CODE REVIEW

File: msgpack_numpy.py [hunk 1]
```diff
@@ -154,37 +159,33 @@ def __init__(self, file_like=None, read_size=0, use_list=None,
                                            max_buffer_size=max_buffer_size,
                                            ext_hook=ext_hook)
 
-def pack(o, stream, default=encode, **kwargs):
+def pack(o, stream, **kwargs):
     """
     Pack an object and write it to a stream.
     """
```
Comment (tiagocoutinho) [inline msgpack_numpy.py]: actually, I think msgpack_numpy would not need to patch `pack()`, `packb()`, `unpack()` and `unpackb()` anymore.

**Comment by lebedov** [comment]
Many of the unit tests appear to be broken by your PR. Can you please fix?

**Comment by tiagocoutinho** [comment]
Yes, of course. Sorry for that.
It should be ok now.

File: msgpack_numpy.py (hunk 1 above)
Comment (tiagocoutinho) [reply msgpack_numpy.py]: Well, I just had a quick look at the msgpack code so... maybe not :-(

**Comment by lebedov** [comment]
Thanks!

**Pull request merged**